### PR TITLE
backend diagram draft 1

### PR DESCRIPTION
**Pull request by annagav**

### CODE REVIEW

**Comment by pdpinch** [comment]
FYI, I tried to look at this on the github web site and it attempted to render the svg, but I think some text is missing. I'll review locally.

**Comment by pdpinch** [comment]
@annagav I'll be taking a look at this, but is there anyone else you want to have review it? Please feel free to tag them.

**Comment by pdpinch** [comment]
What's the relationship between Assignment and User?

File: docs/backend1.svg [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+       xml:space="preserve"
+       style="font-style:normal;font-weight:normal;font-size:15px;line-height:125%;font-family:sans-serif;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1"
+       x="895"
+       y="832.36218"
+       id="text4734"
+       sodipodi:linespacing="125%"><tspan
+         sodipodi:role="line"
+         id="tspan4736"
+         x="895"
+         y="832.36218">token</tspan></text>
```
Comment (pdpinch) [inline docs/backend1.svg:680]: Do you know what this token field is for?

File: docs/backend1.svg [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+       xml:space="preserve"
+       style="font-style:normal;font-weight:normal;font-size:15px;line-height:125%;font-family:sans-serif;letter-spacing:0px;word-spacing:0px;fill:#000000;fill-opacity:1;stroke:none;stroke-width:1px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1"
+       x="890"
+       y="799.36218"
+       id="text4730"
+       sodipodi:linespacing="125%"><tspan
+         sodipodi:role="line"
+         id="tspan4732"
+         x="890"
+         y="799.36218">assignmentName</tspan></text>
```
Comment (pdpinch) [inline docs/backend1.svg:669]: Why is assignmentName here and in the assignment table?

**Comment by pdpinch** [comment]
is assignment.data the same as StudentAssignment.data -- or does StudentAssignment.data contain the assignment data + student's state?

**Comment by pdpinch** [comment]
Seems like StudentAssignment needs a foreign key to user?

**Comment by annagav** [comment]
data looks like this:

``` javascript
{
        id: 'scb_ex3',
        name: 'Exercise 3',
        course: 'SCB_SampleExercises',
        course_name: 'Sample Exercises',
        description: 'StarCellBio Exercise 3',
        notebook: {},
        experiments: {},
        ...
}
```

Data is the same for both Assignment and StudentAssignment, except that for StudentAssignment student experiments are saved into the 'experiments' attribute.

**Comment by pdpinch** [comment]
@annagav should we just close this? If it isn't wrong, it's at least missing all the assignment builder models you've added.

**Comment by annagav** [comment]
Assignment builder is separate from starcellbio backend, so this diagram is still correct.

**Comment by pdpinch** [comment]
Ok, this is better than nothings. Let's just merge it.

**Pull request merged**